tests/helpers/kube.go

```diff
@@ -1429,6 +1501,132 @@ func WaitForNodeConditionWithCheckName(
 	}, EventuallyWaitTimeout, WaitInterval, "node %s should have a condition with check name %s", nodeName, checkName)
 }
 
+// WaitForNodeConditionHealthy waits for the node condition to become healthy
+// (Status=False, Reason=checkName+"IsHealthy").
+// This is needed when testing with real nodes where gpu-health-monitor
+// takes time to update conditions asynchronously.
+func WaitForNodeConditionHealthy(
+	ctx context.Context, t *testing.T, c klient.Client, nodeName, checkName string,
+) {
+	require.Eventually(t, func() bool {
+		node, err := GetNodeByName(ctx, c, nodeName)
+		if err != nil {
+			t.Logf("failed to get node %s: %v", nodeName, err)
+			return false
+		}
+
+		for _, condition := range node.Status.Conditions {
+			if string(condition.Type) == checkName &&
+				condition.Status == v1.ConditionFalse &&
+				condition.Reason == checkName+"IsHealthy" {
+				t.Logf("Found healthy node condition: Type=%s, Reason=%s, Status=%s, Message=%s",
+					condition.Type, condition.Reason, condition.Status, condition.Message)
+
+				return true
+			}
+		}
+
+		t.Logf("Node %s condition '%s' not yet healthy", nodeName, checkName)
+
+		return false
+	}, EventuallyWaitTimeout, WaitInterval, "node %s should have healthy condition %s", nodeName, checkName)
+}
+
+// WaitForNodeConditionUnhealthy waits for the node condition to become unhealthy
+// (Status=True, Reason=checkName+"IsNotHealthy").
+// This is needed when testing with real nodes where gpu-health-monitor
+// takes time to update conditions asynchronously.
+func WaitForNodeConditionUnhealthy(
```

Comment:
can we re-use WaitForNodeConditionWithCheckName or modify that function itself to fit the usecase above as well?